### DISCUSSION TITLE
Argument of type '{ id: Number; }' is not assignable to parameter of type 'FilterQuery<Post>

### DISCUSSION THREAD

**Discussion by adetorodev** [MichalLytek/type-graphql#1246] [Q&A]
**Argument of type '{ id: Number; }' is not assignable to parameter of type 'FilterQuery<Post>'.**
I want to filter my Post by id

**To Reproduce**
```
@Query(() => Post, {nullable: true})
  post(
    @Arg('identifier') id: Number,
    @Ctx() { em }: myContext): Promise<Post | null> {
    return em.findOne(Post, {id});
  }
}
```

**Logs**
This is the logged error.

```
(property) id: Number
Argument of type '{ id: Number; }' is not assignable to parameter of type 'FilterQuery<Post>'.
  Types of property 'id' are incompatible.
    Type 'Number' is not assignable to type 'FilterValue<number> | undefined'.
      Type 'Number' is missing the following properties from type 'FilterValue2<number>[]': length, pop, push, concat, and 29 more.ts(2345)
```

**Environment (please complete the following information):**

- OS: [e.g. Linux(Ubuntu]
- Node (17)
- Package version [1.1.1] (please check if the bug still exist in newest release)
- TypeScript version (4.6.3)
- 
![Screenshot from 2022-03-25 21-28-23](https://user-images.githubusercontent.com/39957559/160219326-fb0fcc82-7c23-4c5c-84a1-a46c53cac907.png)

**Comment by MichalLytek** [comment]
`Number` is not `number`

**Reply by MichalLytek** [reply]
And it's not a TypeGraphQL issue, you should ask on stack overflow
